packages/material-ui/src/MenuItem/MenuItem.js

```diff
@@ -28,28 +54,72 @@ export const styles = (theme) => ({
     ...theme.typography.body2,
     minHeight: 'auto',
   },
+  /* Styles applied to a Menu Item's children when a subMenu is present */
+  subMenuItemWrapper: {
+    width: '100%',
+    display: 'flex',
+    justifyContent: 'space-between',
+  },
+  /* Styles applied to the subMenuIcon when it is present */
+  subMenuIcon: {
+    marginLeft: theme.spacing(2),
+  },
+  /* Styles applied to subMenuIcon when dirction is 'rtl' */
+  rtlSubMenuIcon: {
+    transform: 'rotate(-180deg)',
+  },
 });
 
 const MenuItem = React.forwardRef(function MenuItem(props, ref) {
+  const theme = useTheme();
+
   const {
+    children,
     classes,
     className,
     component = 'li',
     disableGutters = false,
+    handleArrowRightKeydown,
+    key,
     ListItemClasses,
+    openSubMenu = false,
+    onKeyDown,
     role = 'menuitem',
     selected,
+    subMenu,
+    subMenuIcon: SubMenuIcon = KeyboardArrowRight,
+    setParentOpenSubMenuIndex,
     tabIndex: tabIndexProp,
+    handleParentClose,
     ...other
   } = props;
 
+  const listItemRef = React.useRef(null);
+  const handleOwnRef = React.useCallback((instance) => {
+    // #StrictMode ready
+    listItemRef.current = ReactDOM.findDOMNode(instance);
```

Comment:
We no longer support non-ref-forwarding class components (see https://github.com/mui-org/material-ui/pull/21811). `handleOwnRef = listItemRef`.

Reply:
I think this is addressed in my latest commits. Let me know if not.